go/mcap/indexed_message_iterator.go

```diff
@@ -136,10 +130,34 @@ func (it *indexedMessageIterator) loadNextChunk() error {
 	if err != nil {
 		return fmt.Errorf("failed to parse chunk: %w", err)
 	}
+	// decompress the chunk data
+	var chunkData []byte
+	switch CompressionFormat(parsedChunk.Compression) {
+	case CompressionNone:
+		chunkData = parsedChunk.Records
+	case CompressionZSTD:
+		reader, err := zstd.NewReader(bytes.NewReader(parsedChunk.Records))
+		if err != nil {
+			return fmt.Errorf("failed to read zstd chunk: %w", err)
+		}
+		defer reader.Close()
+		chunkData, err = io.ReadAll(reader)
+		if err != nil {
+			return fmt.Errorf("failed to decompress zstd chunk: %w", err)
+		}
+	case CompressionLZ4:
+		reader := lz4.NewReader(bytes.NewReader(parsedChunk.Records))
```

Comment:
reusing these readers across chunks can provide a speedup - also does this one need to be closed if not reused?

Reply:
This is just moved code, but I can take a look at this 👍

Reply:
Yep - now reusing decoders for both zstd and lz4. Please note that:
* lz4 has no `Close()` functionality for its [Reader](https://pkg.go.dev/github.com/pierrec/lz4#Reader)
* zstd decoder has `Close()`, but the [documentation states](https://pkg.go.dev/github.com/klauspost/compress/zstd#Decoder.Close) that the decoder cannot be reused after `Close()` is called. I've taken the `defer Close()` out as a result.